### PR TITLE
APL-1560 refactoring Vote component

### PR DESCRIPTION
**Pull request by andrii-firstbridge**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    @Inject
+    public VoteTable(PollTable pollTable) {
+        super("vote", VOTE_LONG_KEY_FACTORY);
+        this.pollTable = pollTable;
+    }
+
+
+    @Override
+    public void save(Connection con, Vote entity) throws SQLException {
+        Blockchain blockchain = CDI.current().select(Blockchain.class).get();
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java:45]: Incorrect implementation. see as example 'CurrencyTransferTable'.
The height should be assigned outside 'Table', usually inside 'service' class before storing entity.
`...CDI.current().select(Blockchain.class).get()` should NOT be used here.

In Service you can use:  BlockChainInfoService blockChainInfoService;  for getting current height.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetPollVote.java [hunk 2]
```diff
@@ -28,14 +28,17 @@
...
 import com.apollocurrency.aplwallet.apl.util.JSON;
 import org.json.simple.JSONStreamAware;
 
 import javax.enterprise.inject.Vetoed;
+import javax.enterprise.inject.spi.CDI;
 import javax.servlet.http.HttpServletRequest;
 
 @Vetoed
 public class GetPollVote extends AbstractAPIRequestHandler {
+    private static final PollService pollService = CDI.current().select(PollService.class).get();
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetPollVote.java:41]: You can put that into : AbstractAPIRequestHandler for common usage in different places / handlers.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetPollVotes.java [hunk 3]
```diff
@@ -20,37 +20,40 @@
...
 import org.json.simple.JSONObject;
 import org.json.simple.JSONStreamAware;
 
 import javax.enterprise.inject.Vetoed;
+import javax.enterprise.inject.spi.CDI;
 import javax.servlet.http.HttpServletRequest;
 
 @Vetoed
 public class GetPollVotes extends AbstractAPIRequestHandler {
+    private static final PollService pollService = CDI.current().select(PollService.class).get();
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetPollVotes.java:42]: The same. Put that into 'AbstractAPIRequestHandler' class

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetState.java [hunk 4]
```diff
@@ -20,29 +20,26 @@
...
+import javax.enterprise.inject.spi.CDI;
 import javax.servlet.http.HttpServletRequest;
 import java.net.InetAddress;
 
+import static com.apollocurrency.aplwallet.apl.core.transaction.TransactionType.lookupCurrencyExchangeOfferFacade;
+
 @Deprecated
 @Vetoed
 public final class GetState extends AbstractAPIRequestHandler {
+    private static final PollService pollService = CDI.current().select(PollService.class).get();
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetState.java:42]: The same as above.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
+package com.apollocurrency.aplwallet.apl.core.dao.state.poll;
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java:1]: Apollo Copyright is missing

File: apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/service/state/impl/PollServiceImplTest.java [hunk 6]
```diff
@@ -6,6 +6,7 @@
 import com.apollocurrency.aplwallet.apl.core.converter.rest.IteratorToStreamConverter;
 import com.apollocurrency.aplwallet.apl.core.dao.state.poll.PollResultTable;
 import com.apollocurrency.aplwallet.apl.core.dao.state.poll.PollTable;
+import com.apollocurrency.aplwallet.apl.core.dao.state.poll.VoteTable;
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/service/state/impl/PollServiceImplTest.java:9]: Apollo Copyright is missing

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    private static final LongKeyFactory<Vote> VOTE_LONG_KEY_FACTORY = new LongKeyFactory<>("id") {
+        @Override
+        public DbKey newKey(Vote vote) {
+            return vote.getDbKey();
+        }
+    };
+    private PollTable pollTable;
+
+    @Inject
+    public VoteTable(PollTable pollTable) {
```
[the remark this answers is not in the document]
Comment (andrii-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java:37]: I need PollTable here, so I have ti use Inject.

**Pull request merged**